Overhauling docs

Domi250 opened the pull request:
# Overhauling docs
I want to look at all the doc files and improve them.

## Description
In my opinion, parts of the current doc are confusingly structured or incomplete. I want to go over all the doc files, find those parts, and improve them.

I created the PR this early in order to share my suggested changes early.

Silverarmor commented:
Hey @Domi250 thanks for your work here. Are you expecting to do any further edits or shall I merge this?

(Also are you on the Discord server? Please @ me or reach out in the dev channels)

Domi250 commented:
Don't have much time atm. I've done a new README, with both advantages and disadvantages. Let me know what you think of it and if I should change something.

Silverarmor commented on README.md:
```diff
@@ -1,62 +1,48 @@
+Refer to our [Installation Guide](https://spotdl.rtfd.io/en/latest/installation/) for more details.
 
-- Python (**Recommended**)
+### Python (Recommended)
```
can this be "Recommended Method" or similar?

Domi250 replied:
Good idea

Silverarmor commented on README.md:
```diff
@@ -84,27 +79,43 @@ We recommend the above option, but if you want to install FFmpeg system-wide,
 
-To get started right away:
-
+Without options:
```
this isnt the best phrasing. Could you think of an alternative?

Domi250 replied:
I could make it longer and more descriptive:
"Using SpotDL without options:"
Or what direction did you have in mind?

Silverarmor commented on mkdocs.yml:
```diff
@@ -8,9 +8,9 @@ site_dir: "site"
 
 nav:
 - Home: index.md
+- Installation: installation.md
```
Great change!

Silverarmor commented on README.md:
````diff
@@ -84,27 +79,43 @@ We recommend the above option, but if you want to install FFmpeg system-wide,
 ```
 
+There are different **operations** spotDL can perform. The *default* is `download`, which simply downloads the songs from YouTube and embeds metadata.
+The **query** is usually list of Spotify URLs, but for other operations it might be a list of files.
````
a query could also be just one link - and in the case of `sync` usually is - could you make this clearer?

Domi250 replied:
How about:

The **query** is usually a Spotify URL, but for other operations it might be a filename. It is also possible to provide multiple queries at once.

Silverarmor reviewed (changes requested):
Great work! A couple questions here and I've left a few review comments 

any reason we are remoivng
 - mit license and # of commits since last release badges
 - prereqs (vc++ 2019 redist and py3.7
 - msg abt YTM must be avail for spotDL to work

xnetcat commented:
I've merged because it was blocking v4.1.0 release. If you want add additional changes to documentation feel free to open new PR for v4.1.1

Domi250 commented:
Regarding your other Questions:
> any reason we are remoivng
> 
>     * mit license and # of commits since last release badges
The used MIT license tag was no longer supported so I removed it, but I see xnetcat found a new one :)
I removed the commits bc I thought most people don't care about them and too many tags cause people too ignore all of them.
> 
>     * prereqs (vc++ 2019 redist and py3.7
I wanted to keep the installation description in the Readme short, and provide a link to the docs with a more detailed guide. I also had the impression that many people on windows already have those installed and can therefore ignore the python and vc++ steps. But I've never installed spotdl on windows - if those aren't pre-installed, it might be better to keep them in there.
> 
>     * msg abt YTM must be avail for spotDL to work
Mainly, because there are ways around this (VPN or yt as audio provider) which means it's not it's not a real requirement. Also, most people probably don't know if yt music is available and therefore won't remember this warning. It makes more sense to have it in the troubleshooting section.
